### PR TITLE
Have 'Segment.chunks' yield smaller 'Segment's

### PR DESCRIPTION
**Pull request by bessman**
Per discussion in #35.

Three tests are failing so it wasn't quite as straight-forward as I'd hoped. I'll try to get them passing.

### CODE REVIEW

**Comment by bessman** [comment]
Tests are passing. I reverted the changes to `_Chunk`, since it's not longer needed except for compatibility.

Should `Segment` have a `__len__` instead?

**Comment by bessman** [comment]
I'm working on updating outdated docstrings referring to `Chunk`.

**Comment by bessman** [comment]
One subtle change that comes with this is that `Segment` instances can now be immediately adjacent to each other. Previously, one could rely on data immediately before or after a `Segment` being null. Could that be a problem?

**Comment by eerimoq** [comment]
Will think about that tomorrow.

**Comment by eerimoq** [comment]
I think it's fine.

File: bincopy.py [hunk 1]
```diff
@@ -351,13 +351,8 @@ class Segment:
 
     """
 
-    class Chunk(namedtuple("Chunk", ["address", "data"])):
+    _Chunk = namedtuple("Chunk", ["address", "data"])
```
Comment (eerimoq) [inline bincopy.py:354]: Remove _Chunk? Should be fine. Very few users should use it. =)

**Comment by bessman** [comment]
Went ahead and added `Segment.__len__`, taking `word_size_bytes` into account.

**Pull request merged**